streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -402,8 +402,8 @@ public void shouldReturnThreadMetadata() {
         assertNotNull(threadMetadata);
         assertEquals(2, threadMetadata.size());
         for (ThreadMetadata metadata : threadMetadata) {
-            assertTrue("#threadState() was: " + metadata.threadState() + "; expected either RUNNING or CREATED",
-                Utils.mkList("RUNNING", "CREATED").contains(metadata.threadState()));
+            assertTrue("#threadState() was: " + metadata.threadState() + "; expected either RUNNING, PARTITIONS_REVOKED, PARTITIONS_ASSIGNED, or CREATED",
```

Comment:
A few questions:

1. Why we cannot use the `streams` member of the test class itself since it seems the same as the one created here (i.e. empty topology)?

2. What is the `METRICS_RECORDING_LEVEL_CONFIG` set to INFO used for?

3. For the metadata.threadState() itself, since it could be any values except `PEDNING_SHUTDOWN` and `DEAD` does it worth to check it at all?

4. I'm wondering if it is better to fill into the topology with some processor and then check the following `active` and `standby` tasks as non-empty would cover more scenarios? In addition, we can use `waitForCondition` to wait for the state to become `RUNNING` and the tasks list become non-empty.

Reply:
I just fixed the test... Original author was @fhussonnois . Maybe he can elaborate. I personally did not think about the test itself...

(1) because we modify the config, but I don't know why we need to do this (cf. your question 2)
(2) no idea
(3) well, the test as-is is rather basic, so might be ok
(4) we can always make test more sophisticated (maybe we should have done this in the original PR...) can still do this (but we might want to create a JIRA, than? Maybe Florian is interesting helping out here?)

Reply:
Regarding 1) and 2), I think it is not needed, and hence we can use the same `streams` object.
3) fair enough.
4) sounds good, keep it as is then.

Reply:
Created https://issues.apache.org/jira/browse/KAFKA-6381 to track this.